lib/puppet/indirector/file_server.rb

```diff
@@ -29,6 +29,7 @@ def find(request)
     return nil unless path = mount.find(relative_path, request)
     result = model.new(path)
     result.links = request.options[:links] if request.options[:links]
+    result.checksum_type = request.options[:checksum_type] if request.options[:checksum_type]
```

Comment:
After reviewing this fix and the fix for PUP-3888, would it be possible to set these options only once in a helper method, perhaps in `Puppet::FileServing::TerminusHelper`?  Seem like there's four places where we're setting both `links` and `checksum_type` (the two in this PR, the one in the fix for PUP-3888, and one already present in `Puppet::FileServing::TerminusHelper`).